### PR TITLE
Implement ``numeric_only`` support for ``DataFrame.quantile``

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -2959,11 +2957,18 @@ def quantile(self, q=0.5, axis=0, numeric_only=no_default, method="default"):
...
+                _dask_method_name="quantile",
             )
         else:
             _raise_if_object_series(self, "quantile")
-            num = self._get_numeric_data()
-            quantiles = tuple(quantile(self[c], q, method) for c in num.columns)
+            num = (
+                self
+                if PANDAS_GT_200
+                and not numeric_kwargs.get("numeric_only", False)
```
Comment (j-bennet) [inline dask/dataframe/core.py:2967]: This is an odd-looking condition, it's hard to read:

```python
and not numeric_kwargs.get("numeric_only", False)
or not numeric_kwargs.get("numeric_only", True)
```

Are you looking for `numeric_only in (False, no_default)`?

**Review by j-bennet — APPROVED** [review]
Mostly 👍 , one small question.

File: dask/dataframe/core.py (hunk 1 above)
Comment (phofl) [reply dask/dataframe/core.py]: No not that simple unfortunately. We have to adjust for the changed default in pandas. 

I rewrote the condition, let me know whether this is better

Comment (j-bennet) [reply dask/dataframe/core.py]: How about this? Just a little bit simpler.

```
num = (
    self.get_numeric_data()
    if numeric_only is True
    or (not PANDAS_GT_200 and numeric_only is no_default)
    else self
)
```

Comment (phofl) [reply dask/dataframe/core.py]: changed, thx

**Review by j-bennet — APPROVED** [review]
👍 

Nitpick: if the PR template checkboxes don't make sense to you, I would suggest to delete them from PR, rather than leave unchecked. I mean these:

```
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes pre-commit run --all-files
```

In the list of PRs, unchecked boxes create the wrong impression that you're still working on it:

image

**Comment by phofl** [comment]
Will do in the future

**Pull request merged**